addon/utils/recursive-keys.js

```diff
@@ -0,0 +1,48 @@
+import isObject from 'ember-changeset/utils/is-object';
+import { isArray } from '@ember/array';
+
+/**
+ * Recursivly loops over all sub object and
```

Comment:
Let's add something about how this function only returns child-most (or leaf-most) keys. Example:

```js
/**
 * Given an object, return an array of the leaf-most keys
 * in nested string format.
 */
```